jetty-http/src/main/java/org/eclipse/jetty/http/MultiPartFormInputStream.java

```diff
@@ -492,9 +497,6 @@ protected void parse()
         Handler handler = new Handler();
```

Comment:
To save an unnecessary object creation, only make the Handler if you're going to parse.

Reply:
This is just below the check where we decide to parse or not, so this is only created once we have decided to parse.